components/signin/public/identity_manager/brave_identity_manager.h

```diff
@@ -0,0 +1,63 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_SIGNIN_PUBLIC_IDENTITY_MANAGER_BRAVE_IDENTITY_MANAGER_H_
+#define BRAVE_COMPONENTS_SIGNIN_PUBLIC_IDENTITY_MANAGER_BRAVE_IDENTITY_MANAGER_H_
+
+#include <memory>
+
+#include "components/signin/public/identity_manager/identity_manager.h"
+
+// Purpose of this class is to redirect to empty implementation
+// syncer::ProfileSyncService::identity_manager_
+// These must be overridden and empty:
+//    GetAccountsInCookieJar
+//    AddObserver
+//    RemoveObserver
+//    GetPrimaryAccountMutator
+
+class GoogleServiceAuthError;
+class BraveAccountFetcherService;
+
+namespace signin {
+
+struct AccountsInCookieJarInfo;
+class PrimaryAccountMutator;
+class BravePrimaryAccountMutator;
+
+class BraveIdentityManager {
```

Comment:
>I don't see the reason why we can't subclass IdentityManager

I to subclass IdentityManager but stumbled with a problem. I can replace all the methods except the constructor.
And I have to call IdentityManager::IdentityManager before initiating BraveIdentityManager - or I don't know the trick how to make it.

In it's turn, `IdentityManager::IdentityManager` calls the methods of these:
```
account_fetcher_service_
diagnostics_provider_
primary_account_manager_
token_service_
```

So it made me to implement stubs classes of `BraveAccountFetcherService`, `brave_primary_account_manager`, `brave_profile_oauth2_token_service`, `BraveAccountsCookieMutator`, `BraveAccountsMutator`, `BraveDiagnosticsProvider`.

Here is a branch of my attempt https://github.com/brave/brave-core/compare/ab_temp_no_identity?expand=1.

I gave up when it required me to make Brave empty stub for some service, dont recall the name, but something huge to subclass and for sure it would pull more in it's ctor.

>Otherwise, this will get broken easily when ProfileSyncService and SyncAuthManager access more than these methods and we need to make sure other parts of IdentityManager still work other than the part we stubbed.

Yes.
I would go with either this option or use 2 lines patch in ProfileSyncService.

Reply:
Isn't it just something like this?
```
BraveIdentityManager::BraveIdentityManager(IdentityManager::InitParameters&& parameters)
   : IdentityManager(parameters) {}
```
And for `IdentityManager::InitParameters`, please take a look at `IdentityManagerFactory`, `identity_manager_builder` will do the job and just stub the thing we need like `PrimaryAccountMutator`

Reply:
because it required almost all the members of `IdentityManagerFactory` to be stubbed, and some of them pull more dependencies, which also must be stubbed.

Reply:
Redone completely in 4eff8e52e600de280c32a4bcbf7e0ff2a752f546 .